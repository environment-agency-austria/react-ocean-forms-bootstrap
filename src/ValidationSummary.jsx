/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'reactstrap';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faExclamationCircle from '@fortawesome/fontawesome-free-solid/faExclamationCircle';
import { ValidationSummary } from 'react-ocean-forms';

/**
 * Component for displaying a summary of all
 * validation errors of the form this component
 * lives in.
 */
function BootstrapValidationSummary(props) {
  const {
    title,
    ...rest
  } = props;

  return (
    <ValidationSummary
      title={title}
      {...rest}
      renderFieldError={(id, fieldName, errors, linkCallback) => (
        <li key={id}>
          <FontAwesomeIcon icon={faExclamationCircle} className="mr-2" />
          <a href="#" onClick={linkCallback}>
            {fieldName}: {errors}
          </a>
        </li>
      )}

      render={children => (
        <Alert color="danger" className="validation-summary">
          {children}
        </Alert>
      )}
    />
  );
}

BootstrapValidationSummary.defaultProps = {
  title: 'ojs_form_validationSummaryHeader',
};

BootstrapValidationSummary.propTypes = {
  title: PropTypes.string,
};
export default BootstrapValidationSummary;
