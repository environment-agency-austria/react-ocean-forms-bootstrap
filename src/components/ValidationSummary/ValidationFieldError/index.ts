export * from './ValidationFieldError';
export * from './ValidationFieldError.types';
