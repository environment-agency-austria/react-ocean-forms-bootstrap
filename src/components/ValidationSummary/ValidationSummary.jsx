/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'reactstrap';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faExclamationCircle from '@fortawesome/fontawesome-free-solid/faExclamationCircle';
import { ValidationSummary as CoreValidationSummary } from 'react-ocean-forms';

/**
 * Component for displaying a summary of all
 * validation errors of the form this component
 * lives in.
 */
function ValidationSummary(props) {
  const {
    title,
    ...rest
  } = props;

  return (
    <CoreValidationSummary
      title={title}
      {...rest}
      renderFieldError={(id, fieldName, errors, linkCallback) => (
        <li key={id}>
          <FontAwesomeIcon icon={faExclamationCircle} className="mr-2" />
          <a href="#" onClick={linkCallback}>
            {fieldName}
            :
            {' '}
            {errors}
          </a>
        </li>
      )}

      render={children => (
        <Alert color="danger" className="validation-summary">
          {children}
        </Alert>
      )}
    />
  );
}

ValidationSummary.displayName = 'ValidationSummary';

ValidationSummary.defaultProps = {
  title: 'ojs_form_validationSummaryHeader',
};

ValidationSummary.propTypes = {
  title: PropTypes.string,
};
export default ValidationSummary;
