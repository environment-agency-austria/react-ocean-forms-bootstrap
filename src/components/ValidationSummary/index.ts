export * from './ValidationSummary';
export * from './ValidationSummary.types';
export * from './ValidationFieldError';
