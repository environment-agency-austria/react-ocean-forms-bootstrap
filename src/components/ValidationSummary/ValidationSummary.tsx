/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import * as React from 'react';

import { ValidationSummary as CoreValidationSummary } from 'react-ocean-forms';
import { Alert } from 'reactstrap';

import { IValidationSummaryProps } from './ValidationSummary.types';
import { ValidationFieldError } from './ValidationFieldError';

/**
 * Component for displaying a summary of all
 * validation errors of the form this component
 * lives in.
 */
export class ValidationSummary extends React.Component<IValidationSummaryProps> {
  public static displayName: string = 'ValidationSummary';

  private renderFieldError = (
    id: string,
    fieldName: string,
    errors: React.ReactNode,
    linkCallback: React.MouseEventHandler,
  ): JSX.Element => {
    const { fieldErrorComponent: FieldErrorComponent = ValidationFieldError } = this.props;

    return (
      <FieldErrorComponent
        id={id}
        fieldName={fieldName}
        errors={errors}
        linkCallback={linkCallback}
      />
    );
  }

  private renderSummary = (children: JSX.Element): JSX.Element => {
    return (
      <Alert color="danger" className="validation-summary">
        {children}
      </Alert>
    );
  }

  public render(): JSX.Element {
    return (
      <CoreValidationSummary
        {...this.props}
        renderFieldError={this.renderFieldError}
        render={this.renderSummary}
      />
    );
  }
}
