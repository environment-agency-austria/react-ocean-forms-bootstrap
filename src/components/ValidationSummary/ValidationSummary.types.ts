import * as React from 'react';
import { PropsOf, Subtract, ValidationSummary } from 'react-ocean-forms';

import { IValidationFieldErrorProps } from './ValidationFieldError';

type TUpstreamValidationSummaryProps = PropsOf<typeof ValidationSummary>;

/**
 * Props for the ValidationSummary component
 */
export interface IValidationSummaryProps extends Subtract<TUpstreamValidationSummaryProps, { render: string; renderFieldError: string }> {
  fieldErrorComponent?: React.ComponentType<IValidationFieldErrorProps>;
}
