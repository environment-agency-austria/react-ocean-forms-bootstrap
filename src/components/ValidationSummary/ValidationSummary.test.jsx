import React from 'react';
import { shallow } from 'enzyme';
import { ValidationSummary as CoreValidationSummary } from 'react-ocean-forms';

import ValidationSummary from './ValidationSummary';

describe('<ValidationSummary />', () => {
  const wrapper = shallow((
    <ValidationSummary />
  ));

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('render prop', () => {
    const renderProp = wrapper.find(CoreValidationSummary).prop('render');

    it('should set the render prop correctly', () => {
      expect(typeof renderProp).toBe('function');
    });

    it('should correctly render the render prop', () => {
      const mockChildren = <div id="mock" />;
      const renderWrapper = shallow(renderProp(mockChildren));
      expect(renderWrapper).toMatchSnapshot();
    });
  });

  describe('renderFieldError prop', () => {
    const renderFieldErrorProp = wrapper.find(CoreValidationSummary).prop('renderFieldError');

    it('should set the render prop correctly', () => {
      expect(typeof renderFieldErrorProp).toBe('function');
    });

    it('should correctly render the render prop', () => {
      const renderWrapper = shallow(renderFieldErrorProp(
        'mock-id',
        'mock-name',
        'mock-error',
        () => {},
      ));
      expect(renderWrapper).toMatchSnapshot();
    });
  });
});
