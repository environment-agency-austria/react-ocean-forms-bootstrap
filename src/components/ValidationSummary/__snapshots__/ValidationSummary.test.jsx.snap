// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidationSummary /> render prop should correctly render the render prop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alert
    className="validation-summary"
    closeAriaLabel="Close"
    color="danger"
    fade={true}
    isOpen={true}
    tag="div"
    transition={
      Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      }
    }
  >
    <div
      id="mock"
    />
  </Alert>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "appear": true,
      "baseClass": "fade",
      "baseClassActive": "show",
      "children": Array [
        null,
        <div
          id="mock"
        />,
      ],
      "className": "validation-summary alert alert-danger",
      "enter": true,
      "exit": true,
      "in": true,
      "innerRef": undefined,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "role": "alert",
      "tag": "div",
      "timeout": 150,
      "unmountOnExit": true,
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "mock",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "children": Array [
          null,
          <div
            id="mock"
          />,
        ],
        "className": "validation-summary alert alert-danger",
        "enter": true,
        "exit": true,
        "in": true,
        "innerRef": undefined,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "role": "alert",
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "mock",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ValidationSummary /> renderFieldError prop should correctly render the render prop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <li>
    <FontAwesomeIcon
      border={false}
      className="mr-2"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f06a",
            "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
          ],
          "iconName": "exclamation-circle",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <a
      href="#mock-id"
      onClick={[Function]}
    >
      mock-name
      :
       
      mock-error
    </a>
  </li>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "mock-id",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FontAwesomeIcon
          border={false}
          className="mr-2"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f06a",
                "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
              ],
              "iconName": "exclamation-circle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />,
        <a
          href="#mock-id"
          onClick={[Function]}
        >
          mock-name
          :
           
          mock-error
        </a>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "border": false,
          "className": "mr-2",
          "fixedWidth": false,
          "flip": null,
          "icon": Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f06a",
              "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
            ],
            "iconName": "exclamation-circle",
            "prefix": "fas",
          },
          "inverse": false,
          "listItem": false,
          "mask": null,
          "pull": null,
          "pulse": false,
          "rotation": null,
          "size": null,
          "spin": false,
          "symbol": false,
          "title": "",
          "transform": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "mock-name",
            ":",
            " ",
            "mock-error",
          ],
          "href": "#mock-id",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "mock-name",
          ":",
          " ",
          "mock-error",
        ],
        "type": "a",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "mock-id",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FontAwesomeIcon
            border={false}
            className="mr-2"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f06a",
                  "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
                ],
                "iconName": "exclamation-circle",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />,
          <a
            href="#mock-id"
            onClick={[Function]}
          >
            mock-name
            :
             
            mock-error
          </a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "border": false,
            "className": "mr-2",
            "fixedWidth": false,
            "flip": null,
            "icon": Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f06a",
                "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
              ],
              "iconName": "exclamation-circle",
              "prefix": "fas",
            },
            "inverse": false,
            "listItem": false,
            "mask": null,
            "pull": null,
            "pulse": false,
            "rotation": null,
            "size": null,
            "spin": false,
            "symbol": false,
            "title": "",
            "transform": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "mock-name",
              ":",
              " ",
              "mock-error",
            ],
            "href": "#mock-id",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "mock-name",
            ":",
            " ",
            "mock-error",
          ],
          "type": "a",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ValidationSummary /> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ValidationSummary
    title="ojs_form_validationSummaryHeader"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "render": [Function],
      "renderFieldError": [Function],
      "title": "ojs_form_validationSummaryHeader",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "render": [Function],
        "renderFieldError": [Function],
        "title": "ojs_form_validationSummaryHeader",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
